(#1046) Move asText classes from time package to TextOf

Umbrah opened the pull request:
This is for #1046 
I moved cosntructors from *asText classes in time package to TextOf

0crat commented:
Job [#1123](https://github.com/yegor256/cactoos/issues/1123) is now in scope, role is `REV`

<!-- https://www.0crat.com/footprint/C63314D6Z/2e51bd70-9d44-42eb-92c6-9df1a656094f, version: 0.46.7, hash: ${buildNumber} -->

0crat commented:
This pull request [#1123](https://github.com/yegor256/cactoos/issues/1123) is assigned to @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), here is [why](https://www.0crat.com/footprint/C63314D6Z/698e4d11-fd1f-4662-9604-748535fea803); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @llorllale[/z](https://www.0crat.com/u/llorllale) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

<!-- https://www.0crat.com/footprint/C63314D6Z/4554cc50-b3eb-4ff3-86f8-297061e8ead2, version: 0.46.7, hash: ${buildNumber} -->

vzurauskas commented on src/main/java/org/cactoos/text/TextOf.java:
```diff
@@ -399,6 +403,210 @@ public TextOf(final LocalDate date, final DateTimeFormatter formatter) {
+     * @param locale The locale to use.
+     */
+    public TextOf(final Date date, final String format,
+        final Locale locale) {
```
Line wrapping is inconsistent with other classes, e.g. `FormattedText`. I think we should use the ["bracket should either start/end a line or be paired on the same line"](https://www.yegor256.com/2014/10/23/paired-brackets-notation.html) rule (as in `FormattedText`).

So in this case it would look like this:

```java
public TextOf(
    final Date date,
    final String format,
    final Locale locale
) {
    . . .
}
```

This comment applies to other constructors as well.

vzurauskas commented on src/main/java/org/cactoos/text/TextOf.java:
```diff
@@ -399,6 +403,210 @@ public TextOf(final LocalDate date, final DateTimeFormatter formatter) {
+        super((Scalar<String>) () -> new TextOf(
+            ZonedDateTime.ofInstant(
+                date.toInstant(),
+                ZoneId.of("UTC")
```
Javadoc says system default zone will be used, but this is UTC.

vzurauskas commented on src/main/java/org/cactoos/text/TextOf.java:
```diff
@@ -399,6 +403,210 @@ public TextOf(final LocalDate date, final DateTimeFormatter formatter) {
+     * Formats the milliseconds using the ISO format.
+     * @param milliseconds Milliseconds to format.
+     */
+    public TextOf(final Long milliseconds) {
```
It is very unintuitive that `new TextOf(1234L)` is a textual date. Any reasonable user would expect it to just be a "1234L" string. And if not, why date specifically? A Long number could represent anything: distance, money, temperature, etc. I think formatting milliseconds to ISO date is a responsibility for a new class, `FormattedEpoch` or something like that. And then the no-arg `DateAsText` constructor could be moved there, as now it's lost in the migration.

This comment applies to other constructors taking milliseconds as well.

Umbrah commented:
@vzurauskas I fixed comments according to utc zone and constructors formatting. As for constructors with long parameters, I added to-do to create a class like you described and use it in that ctors.

vzurauskas commented:
@Umbrah I think we should leave the constructors with `Long` arguments in the classes they were in originally rather than moving them back and forth with this task and then the new one somebody is going to implement. It might be a long time until they are in the right place, and they made more sense in those original classes. Once somebody moves them to the new class, they can delete the old ones.

Also, in the task description I would specify that the new class is not a `Date`, but a textual representation of a date, a type of `Text`.

Umbrah commented:
@vzurauskas I agree, I will move them back and just add to-do to original classes. 

> Also, in the task description I would specify that the new class is not a Date, but a textual representation of a date, a type of Text.

I think this new class should represent a Date, so we can work with it like new EpochDate(1000L).value() - returns Date; 
And use this class in constructor of TextOf.
Otherwise we will get the same situation which this PR is supposed to fix - there will be a class representing Text but separate from TextOf and in different package. Please let me know what do you think about it.

vzurauskas commented:
@Umbrah That makes sence, although there already is a class like that in the JDK: [Date](https://docs.oracle.com/javase/7/docs/api/java/util/Date.html#Date(long)). So maybe there’s no need for a new class after all. Perhaps you should just update the todo with these considerations and let the developer who takes that task to decide?

Umbrah commented:
@vzurauskas Agree, we can just use new TextOf(new Date()), no need in separate class. I deleted those logic with millis from TextOf. In this case I don't think we need a need a to-do, maybe let's see what ARC think?

vzurauskas commented:
@Umbrah Alright.

@llorllale Looks good to me. Note the discussion we had about the constructors with millisecond arguments, see if you agree with this.

Umbrah commented:
@llorllale ping

llorllale commented:
@Umbrah @vzurauskas I agree that `new TextOf(long)` should print out the number in textual form.

I think I see a mismatch between part of what was discussed (create todo to add new classes/leave old classes with long parameters behind) and what is actually implemented.

But that's ok - this PR is fine as it is.

llorllale commented:
@rultor merge

rultor commented:
> @rultor merge

@llorllale OK, I'll try to merge now. You can check the progress of the merge [here](http://www.rultor.com/t/18859-491111753)

rultor commented:
> @rultor merge

@llorllale Done! FYI, the full log is [here](http://www.rultor.com/t/18859-491111753) (took me 13min)

0crat commented:
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vzurauskas[/z](https://www.0crat.com/u/vzurauskas), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

<!-- https://www.0crat.com/footprint/C63314D6Z/ba4a96d2-4bc9-4786-8af4-a9a1d40f5a2c, version: 0.46.10, hash: ${buildNumber} -->

0crat commented:
Code review was too long (10 days), architects (@llorllale) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

<!-- https://www.0crat.com/footprint/C63314D6Z/b5c89409-5eed-4da4-8dd3-386ef66fc2cc, version: 0.46.10, hash: ${buildNumber} -->

0crat commented:
The job [#1123](https://github.com/yegor256/cactoos/issues/1123) is now out of scope

<!-- https://www.0crat.com/footprint/C63314D6Z/90ec68df-8994-47d6-b78f-f5219da738c8, version: 0.46.10, hash: ${buildNumber} -->

0crat commented:
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

<!-- https://www.0crat.com/footprint/C63314D6Z/fd880360-ba4a-404e-8c8e-0bdbda7bb76f, version: 0.46.10, hash: ${buildNumber} -->

sereshqua commented:
@0crat quality good

0crat commented:
Order was finished, quality is "good": +20 point(s) just awarded to @vzurauskas[/z](https://www.0crat.com/u/vzurauskas)

<!-- https://www.0crat.com/footprint/C63314D6Z/aa6196d1-1f15-40f8-b51f-8b13bfe6275a, version: 0.46.10, hash: ${buildNumber} -->

0crat commented:
Quality review completed: +4 point(s) just awarded to @sereshqua[/z](https://www.0crat.com/u/sereshqua)

<!-- https://www.0crat.com/footprint/C63314D6Z/82f3c4c7-c1b9-45e1-9303-ddd51b034769, version: 0.46.10, hash: ${buildNumber} -->
